controllers/operator_controller.go

```diff
@@ -400,3 +409,26 @@ func mapBDStatusToReadyCondition(existingBD *rukpakv1alpha1.BundleDeployment, ob
 func isBundleDepStale(bd *rukpakv1alpha1.BundleDeployment) bool {
 	return bd != nil && bd.Status.ObservedGeneration != bd.GetGeneration()
 }
+
+// Generate reconcile requests for all operators affected by a catalog change
+func operatorRequestsForCatalog(ctx context.Context, c client.Reader, logger logr.Logger) handler.MapFunc {
+	return func(object client.Object) []reconcile.Request {
+		// no way of associating an operator to a catalog so create reconcile requests for everything
```

Comment:
If we included the catalog in an `Operator.Status` field could this be filtered that way via a field selector? Probably out of scope for this work, but I definitely think it would be ideal in the future to have some way of saying `Catalog` X was changed so only re-reconcile `Operator` resources that were resolved to a package in `Catalog` X

Reply:
It'd be useful to have this field but that excludes the possibility of cross catalog updates. I can see this happening if a catalog an operator is installed from is deleted and reinstalled, we'd still need to trigger reconcile to catch any upgrades.

Reply:
>excludes the possibility of cross catalog updates

So with this current implementation we are saying "if there is an update in any catalog we will update the operator"? Are we sure that is something we want to do without some form of explicit action from the user side? What if there is a catalog with a package with the same name in another catalog that is actually a completely different package?

These questions are coming from a lack of knowledge on how operator-controller handles that case today and if there have already been discussions around this so apologies in advance if I am rehashing something

Reply:
> if there is an update in any catalog we will update the operator?

No, we're saying we'll resolve, evaluate constraints, and apply the results of resolution. There's definitely more work we need to do to provide sufficient knobs on the Operator API to help admins avoid that sort of problem, but that's a problem for resolution to solve, so IMO, we shouldn't preempt that right now by not even giving the resolver a chance to take a look.

Reply:
Sounds good. Thanks for the clarification!